compiler/rustc_typeck/src/astconv/mod.rs

```diff
@@ -504,15 +504,17 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
                             tcx.ty_error().into()
                         }
                     }
-                    GenericParamDefKind::Const => {
-                        let ty = tcx.at(self.span).type_of(param.def_id);
-                        // FIXME(const_generics_defaults)
-                        if infer_args {
-                            // No const parameters were provided, we can infer all.
-                            self.astconv.ct_infer(ty, Some(param), self.span).into()
+                    GenericParamDefKind::Const { has_default } => {
+                        if !infer_args && has_default {
+                            ty::Const::from_anon_const(tcx, param.def_id.expect_local()).into()
```

Comment:
all of the uses of `Const::from_anon_const` have to be replaced with something that works with structs from other crates.

I think it's easiest to add a query called something like `const_param_default`

Reply:
this is also a good test case ^^

Reply:
I'm not totally sure how to implement this, would it just be looking at the HIR for another crate similar to `type_of`?

Reply:
Alright so I've added a query, but `expect_local` inside of it breaks for external crates in the test I added, not sure how to fix that.